### PR TITLE
Feature/search

### PR DESCRIPTION
**Pull request by m1073496**
## What does this PR do?
- Creates a controlled form for user searches
- Implements logic to render matching movies by title in real-time
- Retains all previous functionality!

## How should it be tested?
 - Make a search!

* Need to implement logic to reasonably handle various kinds of user inputs (lowercase, uppercase, etc.)
* Need to implement logic to search by various keywords (genre, overview, etc.)
* Need to remove search bar from UI on Movie Detail view to prevent accidental search changes
* Need to display a helpful message for bad searches (rework logic that is instead displaying all movies when this happens)

### CODE REVIEW

File: src/search/Form.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    this.state = {
+      searchInput: ''
+    }
+  }
+
+  handleChange = event => {
+    this.setState({ [event.target.name]: event.target.value}, () => {
+      this.props.filterMovies(this.state.searchInput)
+    });
+  }
```
Comment (cbmackintosh) [inline src/search/Form.js:16]: Nice work finding such an elegant solution to this problem. So the SetState method can accept a callback function? Which in this case means it won't change the component's state until filterMovies is finished executing? Might need you to walk me through it one more time.

File: src/app/App.js [hunk 2]
```diff
@@ -45,4 +48,24 @@ export default class App extends Component {
...
+    //Want to search by title, overview, and genre
+    const filteredList = this.state.movies.filter(movie => movie.title.includes(query))
+    // console.log(filteredList)
+    // if(filteredList.length === 0) {
+    //   this.setState({ error: 'No results' })
+    // } else {
+    //   this.setState({ searchResults: filteredList })
+    // }
+    this.setState({ searchResults: filteredList })
+  }
```
Comment (cbmackintosh) [inline src/app/App.js:61]: I think we can make this conditional work for us with a few changes. The way I see it, there are three conditions here:
1.) If filteredList.length === 0 and the search bar is empty, then show all movies
2.) If filteredList.length === 0 and the search bar is NOT empty, then set App state error to show "No results"
3.) If filteredList.length > 0, set App state searchResults to filteredList...

**Review by cbmackintosh — APPROVED** [review]
Great work on paired programming today! We've established core functionality for our Iteration 5 extension - we just need to refine it to improve search validation and error handling. Once that's done, we can focus on other cosmetic improvements!

**Comment by m1073496** [comment]
[Iteration 5: Search Bar --> Controlled Form](https://trello.com/c/itHVMrLJ/27-iteration-5-search-bar-controlled-form)

**Comment by m1073496** [comment]
[Iteration 5: Search Bar --> Controlled Form](https://trello.com/c/itHVMrLJ/27-iteration-5-search-bar-controlled-form)

**Pull request merged**